js/main.js

```diff
@@ -76,10 +88,113 @@ function bookmark(event) {
   if ($bookmarkIcon.getAttribute('src') === 'images/bookmark-plus.png') {
     $bookmarkIcon.setAttribute('src', 'images/bookmark-fill.png');
     data.entries.push(xhr.response);
-    data.entryId++;
+
+    var li = document.createElement('li');
+    li.setAttribute('class', 'row justify-end');
+
+    var img = document.createElement('img');
+    img.setAttribute('class', 'dog column-half margin-right-bottom-desktop');
+    img.setAttribute('src', xhr.response[0].url);
+    li.appendChild(img);
+
+    var div = document.createElement('div');
+    div.setAttribute('class', 'column-half');
+    var p = document.createElement('p');
+    if (xhr.response[0].breeds[0]) {
+      p.textContent = xhr.response[0].breeds[0].name;
+      var p2 = document.createElement('p');
+      p2.setAttribute('class', 'inline breed-info margin-none underline margin-bottom text-align-right-mobile width');
+      p2.textContent = 'Click here to learn more!';
+      p2.setAttribute('id', xhr.response[0].id);
+    } else {
+      p.textContent = 'Unknown';
+    }
+    p.setAttribute('class', 'inline padding-none text-align-right-mobile margin-bottom-none margin-top-none');
+    li.appendChild(div);
+    div.appendChild(p);
+    if (p2) {
+      div.appendChild(p2);
+    }
+    $ul.appendChild(li);
```

Comment:
You should make this into its own function to generate a DOM tree for one dog, and then `return` that DOM tree so you can append it where the function is called.

Reply:
Wait, isn't `dogListView` doing this work already?